backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/IncrementalLayoutEngine.java

```diff
@@ -137,12 +145,160 @@ private void layoutNode(Optional<IDiagramEvent> optionalDiagramElementEvent, Nod
         // resize / change position according to the content
         new ContainmentUpdater().update(node);
 
+        // update the border node once the current node bounds are updated
+        Bounds newBounds = Bounds.newBounds().position(node.getPosition()).size(node.getSize()).build();
+        this.layoutBorderNodes(optionalDiagramElementEvent, node.getBorderNodes(), initialNodeBounds, newBounds, layoutConfigurator);
+
         // recompute the label
         if (node.getLabel() != null) {
             node.getLabel().setPosition(this.nodeLabelPositionProvider.getPosition(node, node.getLabel()));
         }
     }
 
+    /**
+     * Update the border nodes position according to the side length change where it is located.<br>
+     * The aim is to keep the positioning ratio of the border node on its side.
+     *
+     * @param borderNodes
+     *            the border nodes to update
+     * @param initialNodeBounds
+     * @param newNodeBounds
+     * @param layoutConfigurator
```

Comment:
Either you write a proper Javadoc, or you don't write Javadoc at all. The same comment could be applied to pretty much all the methods that you have touched in this class.

Reply:
done